src/Query/DbSet.cs

```diff
@@ -296,7 +299,26 @@ private async Task CommitChangesAddedItemsAsync(HashSet<T> items, bool useAsync)
                         if (hasIdentity)
                         {
                             var id = await ExecuteScalarAsync(command, useAsync);
-                            _tableInfo.IdentityPrimaryKey.SetValue(item, id);
+                            var dynamicMethodArgs = new Type[] { typeof(T), typeof(object) };
+                            var dynamicMethod = new DynamicMethod("IdentityPrimaryKey_SetValue", typeof(void), dynamicMethodArgs);
+                            ILGenerator ilGenerator = dynamicMethod.GetILGenerator();
+
+                            ilGenerator.Emit(OpCodes.Ldarg_0);
+                            ilGenerator.Emit(OpCodes.Ldarg_1);
+
+                            if (_tableInfo.IdentityPrimaryKey.PropertyInfo.PropertyType.IsValueType)
+                            {
+                                ilGenerator.Emit(OpCodes.Unbox_Any, _tableInfo.IdentityPrimaryKey.PropertyInfo.PropertyType);
+                            }
+
+                            MethodInfo fnGetIdentity = typeof(T).GetProperty(
+                                _tableInfo.IdentityPrimaryKey.Name, BindingFlags.Public | BindingFlags.Instance).GetSetMethod();
+
+                            ilGenerator.Emit(OpCodes.Callvirt, fnGetIdentity);
+                            ilGenerator.Emit(OpCodes.Ret);
+
+                            var invoke = dynamicMethod.CreateDelegate(typeof(_setIdentityPrimaryKeyDelegate)) as _setIdentityPrimaryKeyDelegate;
```

Comment:
Delegate can by cached like in `DynamicMethodModelFactory`. Key will be generated probably from `T`.